name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/Extractor.java

```diff
@@ -344,7 +355,14 @@ public Status apply(ImportAction action, Context context)
             if (portfolio == null)
                 portfolio = context.getPortfolio();
 
-            return action.process(entry, account, portfolio);
+            Status status = action.process(entry, account, portfolio);
+
+            // check if message was set in DetectDuplicatesAction
+            if (status.getMessage() != null && status.getMessage().equals(ImportAction.IMPORT_SAVINGS_PLAN_ITEM))  
+            { 
+                this.savingsPlanItem = true;
+            }
```

Comment:
The ```Extractor.Item``` has already a property ```data```. I wonder if it makes sense to change ```data``` to something like ```Map<String, Object>```  and attach the detected investment plan transaction to the item itself. Then we do not have to re-search for the transaction later.

What do you think of passing the ```item``` also to the ```ImportAction#process``` methods? The ```DetectDuplicateAction``` can attach it, and the ```InsertAction``` can replace the transaction.

Reply:
I extended `data` the way proposed, but didn't use it. To pass the `item` from and to `process` methods would require too many changes for this special case. The attached message is now used in a bit different way to check for an item to replace an investment plan item.

Reply:
Just reverted the `data` object, because the JSON PDF test cases went wrong...